### ISSUE TITLE
stable urls for grouped objects

### ISSUE THREAD

**Issue by EdwardTown1** [ycba-cia/blacklight-collections2#255]
Hi @yulgit1 yesterday @edgartdata and I spoke w/ Prints and Drawings who drew our attention to and issue regarding URLS for groups of objets (i.e. a search on a particular artist) that predate Blacklight (i.e. were made in VuFind) no longer work. This renders parts of their research guides obsolete. Please may may we think about this issue and discuss at our next meeting.

**Comment by robl** [comment]
Yes, aside from URLs for individual object records (for which it was possible to put systematic redirects in place using known unique identifiers and otherwise consistent paths), any other legacy URLs from VuFind will need to be recreated (for example, by running the same search) in Blacklight, and then either adding a one-off redirect for each or updating the URL where the link is actually used. For legacy links from YCBA documents (etc.), the best approach would be to update the URL in the link there, if possible. For links from external places we can't update, we would either need to create one-off redirects or rely on users to do a Blacklight search once they are redirected to the new general landing page.

**Comment by yulgit1** [comment]
I agree with above @robl.  

Also tied in with this is the example of legacy (vufind searches) embedding in wikipedia pages.  There are a few handful that may just be fixed manually given edit access to wikipedia.

https://en.wikipedia.org/wiki/Special:LinkSearch
 
For example try:
https://britishart.yale.edu
https://collections.britishart.yale.edu

**Comment by robl** [comment]
PS, on endpoints for individual object records, see also issue 72:

https://github.com/ycba-cia/blacklight-collections2/issues/72

**Comment by robl** [comment]
And yes re: Wikipedia. I'd suggest that our best bet would be to have someone update the URLs there to the new ones in Blacklight. From what I know of Wikipedia policy, that may (?) need to be someone not affiliated with the YCBA.

**Comment by yulgit1** [comment]
#72 was basically what was implemented and used now for the vufind->blacklight URL mapping by ID

**Comment by robl** [comment]
Yep; and to elaborate a little in re: "stable URLs for grouped objects," https://github.com/ycba-cia/blacklight-collections2/issues/72 was possible for individual objects because those endpoints could be remapped consistently by means of object IDs; redirects for query-defined object groups would be one-offs because VuFind and Blacklight syntax are so different.

Which is to say, Blacklight URLs for search-results groups are stable, but they are different from the stable URLs for the same searches in VuFind. Noting this just to clarify that this actually a matter of concordance rather than one of stability.

**Comment by edgartdata** [comment]
@robl @yulgit1 Ed and I talked with P&D this week and we told them again that they need to redo in BlackLight their searches for their research guides. We also told them that we would be happy to put redirects on these query-defined object groups once they have them.

**Comment by edgartdata** [comment]
Can we look at Wikipedia examples of legacy vufind ycba links?

**Comment by edgartdata** [comment]
Re. Wikipedia: the YCBA exhibition Drupal links in Wikipedia will need to be considered when the YCBA creates webpages for them in the YCBA website.
The YCBA Vufind saved query links in Wikipedia now get redirected to the YCBA online catalog landing page. We might want to edit these saved queries in Wikipedia.
The YCBA Vufind detailed object records links will need to be edited in Wikipedia at some point. They currently get redirected to their BlackLight counterparts.
